tests/e2e/configurer/cmd.go

```diff
@@ -12,80 +12,76 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/require"
 
 	"github.com/osmosis-labs/osmosis/v7/tests/e2e/chain"
 	"github.com/osmosis-labs/osmosis/v7/tests/e2e/util"
 )
 
-func (s *IntegrationTestSuite) connectIBCChains(chainA *chain.Chain, chainB *chain.Chain) {
-	s.T().Logf("connecting %s and %s chains via IBC", chainA.ChainMeta.Id, chainB.ChainMeta.Id)
-
+func (bc *BaseConfigurer) CreatePool(chainId string, valIdx int, poolFile string) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	defer cancel()
 
-	exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
-		Context:      ctx,
-		AttachStdout: true,
-		AttachStderr: true,
-		Container:    s.hermesResource.Container.ID,
-		User:         "root",
-		Cmd: []string{
-			"hermes",
-			"create",
-			"channel",
-			chainA.ChainMeta.Id,
-			chainB.ChainMeta.Id,
-			"--port-a=transfer",
-			"--port-b=transfer",
-		},
-	})
-	s.Require().NoError(err)
+	containerId := bc.valResources[chainId][valIdx].Container.ID
 
+	bc.t.Logf("running create pool on chain id: %s with container: %s", chainId, containerId)
 	var (
 		outBuf bytes.Buffer
 		errBuf bytes.Buffer
 	)
 
-	err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
-		Context:      ctx,
-		Detach:       false,
-		OutputStream: &outBuf,
-		ErrorStream:  &errBuf,
-	})
-	s.Require().NoErrorf(
-		err,
-		"failed connect chains; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
-	)
-
-	s.Require().Containsf(
-		errBuf.String(),
-		"successfully opened init channel",
-		"failed to connect chains via IBC: %s", errBuf.String(),
+	require.Eventually(
+		bc.t,
+		func() bool {
+			exec, err := bc.dockerPool.Client.CreateExec(docker.CreateExecOptions{
+				Context:      ctx,
+				AttachStdout: true,
+				AttachStderr: true,
+				Container:    containerId,
+				User:         "root",
+				Cmd: []string{
+					"osmosisd", "tx", "gamm", "create-pool", fmt.Sprintf("--pool-file=/osmosis/%s", poolFile), fmt.Sprintf("--chain-id=%s", chainId), "--from=val", "-b=block", "--yes", "--keyring-backend=test",
+				},
+			})
+			require.NoError(bc.t, err)
+			err = bc.dockerPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+				Context:      ctx,
+				Detach:       false,
+				OutputStream: &outBuf,
+				ErrorStream:  &errBuf,
+			})
+			require.NoError(bc.t, err)
+			return strings.Contains(outBuf.String(), "code: 0")
+		},
+		time.Minute,
+		time.Second,
+		"tx returned non code 0; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
```

Comment:
I believe once we merge our PRs we can get rid of this boilerplate

Reply:
Seems that you've done that already: https://github.com/osmosis-labs/osmosis/blob/8b1c2c014cfd5da34a1c30f6c287cfadb8354187/tests/e2e/e2e_util_test.go#L72